frontend/packages/console-shared/src/hooks/usePinnedResources.ts

```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { getActivePerspective } from '@console/internal/reducers/ui';
+import { useUserSettingsCompatibility } from './useUserSettingsCompatibility';
+import { PINNED_RESOURCES_LOCAL_STORAGE_KEY } from '../constants';
+
+type PinnedResourcesType = {
+  [perspective: string]: string[];
+};
+
+const PINNED_RESOURCES_CONFIG_MAP_KEY = 'console.pinnedResources';
+
+export const usePinnedResources = (): [string[], (pinnedResources: string[]) => void, boolean] => {
+  const activePerspective = useSelector(getActivePerspective);
+  const [pinnedResources, setPinnedResources, loaded] = useUserSettingsCompatibility<
+    PinnedResourcesType
+  >(PINNED_RESOURCES_CONFIG_MAP_KEY, PINNED_RESOURCES_LOCAL_STORAGE_KEY, {}, true);
+  return [
+    loaded ? pinnedResources[activePerspective] : [],
```

Comment:
Get an error here "Cannot read property 'admin' of undefined".

loaded is true and pinnedResources is undefined, maybe because it waits on the sync? This works locally as workaround:

```suggestion
    loaded && pinnedResources ? pinnedResources[activePerspective] : [],
```

Reply:
![image](https://user-images.githubusercontent.com/14068621/100773461-e862ca00-33ce-11eb-816f-6dbdd54e2579.png)

Reply:
App won't load at all for me. white screens